TeamCode/src/main/java/org/firstinspires/ftc/teamcode/PixyCam.java

```diff
@@ -0,0 +1,170 @@
+package org.firstinspires.ftc.teamcode;
+
+import android.util.Log;
+
+import com.qualcomm.robotcore.hardware.I2cDeviceSynch;
+
+import java.util.Arrays;
+import java.util.Collections;
+
+/**
+ * Created by NoahR on 10/3/17.
+ * Note about Modern Robotics I2C wiring. (Left to right)
+ * 0 - 1 - 2 - 3
+ * 5v - Data - Clock - gnd
+ */
+
+public class PixyCam
+{
+    I2cDeviceSynch m_deviceClient;
+    private byte[] m_registerValues;
+    private int m_dataStartAddress = 0;
+
+    private final int START_ADDRESS = 0x54;
+    private final int READ_LENGTH = 26; // Start read at 0x58, end at 0x58 + 26
+    private final byte[] PIXY_DEVICE_SIGNATURE = {85, (byte) 170, 85, (byte) 170};
+    private final int SIGNATURE_SPACE_SIZE = 10; // Device signature will be in the first 10 bytes
+
+    PixyCam() {}
+    PixyCam( I2cDeviceSynch deviceClient ) { initialize( deviceClient); }
+
+    public void initialize( I2cDeviceSynch deviceClient )
+    {
+        m_deviceClient = deviceClient;
+        m_deviceClient.engage();
+    }
+
+    public byte[] GetRegisterValues() { return m_registerValues; }
+
+    /**
+     * Update the cashed register values
+     * @return whether the operation was successful
+     */
+    public boolean Update()
+    {
+
+        byte[] registerValues = m_deviceClient.read( START_ADDRESS, READ_LENGTH );
+        byte[] signatureSpace = new byte[SIGNATURE_SPACE_SIZE];
+
+        // Only use the first ten bytes for signature scanning
+        for( int iIndex = 0; iIndex < SIGNATURE_SPACE_SIZE; iIndex++)
+        {
+            signatureSpace[iIndex] = (byte) mod(registerValues[iIndex], 256);
+        }
+
+        // If all signature space values are zero then we are not retrieving valid data.
+        int signatureIndex = indexOfSubArray(signatureSpace, PIXY_DEVICE_SIGNATURE);
+        if( signatureIndex > -1 )
+        {
+            m_registerValues = registerValues;
+            m_dataStartAddress = signatureIndex;
+            return true;
+        }
+        else
+            return false;
+    }
+
+    /**
+     * Update the register values. If the values are not available, halt the thread until they are.
+     */
+    public void ForceUpdate()
+    {
+        while(!Update());
+    }
+
+    public int GetDataStartAddress()
+    {
+        return m_dataStartAddress;
+    }
+
```

Comment:
These accessors below could be dangerous. What if another thread calls `Update()` between my calls to `GetXCenter()` and `GetYCenter()`? Since all this data goes together, consider making a helper class and passing back an instance of that class.

    public class PixyObject {
        public int centerX;
        public int centerY;
        public int width;
        public height;
    }

    public class PixyCam {
        ...
        PixyObject GetObject() { // format all the fields here }
        ...
    }